### PR TITLE
feat: dbt monte carlo integration

### PR DESCRIPTION
**Pull request by Hamza442**

### CODE REVIEW

**Review by AbdouSeck — APPROVED** [review]

File: dataeng/resources/warehouse-transforms.sh [hunk 1]
```diff
@@ -89,3 +133,6 @@ then
     dbt test --models $MODEL_SELECTOR $exclude_param $INDIRECT_SELECTION_PARAM --profile $DBT_PROFILE --target $DBT_TARGET --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ ; ret=$?;
     postCommandChecks "test" $ret ;
 fi
+
+# remove the persisted token vault configs
+rm -rf ${WORKSPACE}/vault-config
```
Comment (pwnage101) [inline dataeng/resources/warehouse-transforms.sh:138]: These lines can be removed once you stop writing a vault token to disk

**Review by pwnage101 — CHANGES_REQUESTED** [review]

File: dataeng/resources/warehouse-transforms.sh [hunk 2]
```diff
@@ -65,6 +65,50 @@ fi
...
+# Creating separate token for each job in its workspace
+# By default token is generated in home directory of server
+# When token location is changed using VAULT_CONFIG_PATH vault cli
+# should find the new location of token using the vault config file
+# This is the expected behaviour for vault cli. But vault cli was
+# not working as expected and is not able to locate the new token location
+# Have to explicitly store token in token environment variable so that
+# vault cli can use the newly generated token for each job
+
+export VAULT_TOKEN="$(cat ${WORKSPACE}/vault-config/vault-token)"
```
[the remark this answers is not in the document]
Comment (Hamza442) [reply dataeng/resources/warehouse-transforms.sh:100]: If we directly write it into the VAULT_TOKEN environment variable then the problem of race condition remains as it is. First we went with this approach but this didn't work. It seems like when we directly write into the environment variable each job might be accessing the same variable.

File: dataeng/jobs/analytics/WarehouseTransforms.groovy [hunk 3]
```diff
@@ -67,8 +80,21 @@ class WarehouseTransforms{
                         string('OPSGENIE_HEARTBEAT_CONFIG_KEY', 'opsgenie_heartbeat_config_key')
                     }
                 }
+                wrappers {
+                    timestamps()
+                    credentialsBinding {
+                    usernamePassword('ANALYTICS_VAULT_ROLE_ID', 'ANALYTICS_VAULT_SECRET_ID', 'analytics-vault');
```
[the remark this answers is not in the document]
Comment (Hamza442) [reply dataeng/jobs/analytics/WarehouseTransforms.groovy]: Updated

File: dataeng/jobs/analytics/WarehouseTransforms.groovy [hunk 4]
```diff
@@ -45,12 +46,24 @@ class WarehouseTransforms{
...
+                        }
+                    }
                 }
                 triggers common_triggers(allVars, env_config)
                 environmentVariables {
                     env('OPSGENIE_HEARTBEAT_NAME', env_config.get('OPSGENIE_HEARTBEAT_NAME'))
                     env('OPSGENIE_HEARTBEAT_DURATION_NUM', env_config.get('OPSGENIE_HEARTBEAT_DURATION_NUM'))
                     env('OPSGENIE_HEARTBEAT_DURATION_UNIT', env_config.get('OPSGENIE_HEARTBEAT_DURATION_UNIT'))
+                    env('REMOTE_CONFIG_DECRYPTION_KEYS_VAULT_KV_PATH', allVars.get('REMOTE_CONFIG_DECRYPTION_KEYS_VAULT_KV_PATH'))
+                    env('REMOTE_CONFIG_DECRYPTION_KEYS_VAULT_KV_VERSION', allVars.get('REMOTE_CONFIG_DECRYPTION_KEYS_VAULT_KV_VERSION'))
```
[the remark this answers is not in the document]
Comment (Hamza442) [reply dataeng/jobs/analytics/WarehouseTransforms.groovy]: Updated

**Review by pwnage101 — APPROVED** [review]

**Pull request merged**